pkgs/tools/security/dirbuster/default.nix

```diff
@@ -0,0 +1,99 @@
+{ lib, stdenvNoCC, fetchurl, makeWrapper, copyDesktopItems, makeDesktopItem, binutils, jre, jdk }:
+
+let
+  name = "dirbuster";
+  version = "1.0-RC1";
+
+  binaryDist = stdenvNoCC.mkDerivation rec {
+    pname = "${name}-bin";
+    inherit version;
+
+    src = fetchurl {
+      url = "mirror://sourceforge/dirbuster/DirBuster%20(jar%20%2B%20lists)/${version}/DirBuster-${version}.tar.bz2";
+      sha256 = "1hgbamzlz83qxcl15nic2jr8542dgq0ghsvrqypc4bhsv7a2v0aj";
+    };
+
+    installPhase = ''
+      # Install dependency binaries
+      install -Dt $out/share/java lib/*
+
+      # Install wordlists
+      install -Dt $out/share/${name} *.{txt,jbrofuzz}
+    '';
+  };
+in
+stdenvNoCC.mkDerivation rec {
+  pname = name;
+  inherit version;
+
+  src = fetchurl {
+    url = "mirror://sourceforge/dirbuster/DirBuster%20Source/${version}/DirBuster-${version}-src.tar.bz2";
+    sha256 = "0p210fp6rbfzvf8a03kmcizbg6f22z621808094dgdpl8d1fj578";
+  };
+
+  patches = [
+    # Open the file browser at location of wordlists for convenience
+    ./default-browse-dir.patch
+  ];
+
+  postPatch = ''
+    substituteAllInPlace com/sittinglittleduck/DirBuster/gui/JDialogAdvSetup.java
+    substituteAllInPlace com/sittinglittleduck/DirBuster/gui/JPanelSetup.java
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      desktopName = "OWASP DirBuster";
+      exec = pname;
+      icon = pname;
+      comment = "Web Application Brute Forcing";
+      categories = [ "Network" ];
+    })
+  ];
+
+  buildInputs = [ binaryDist ];
+  nativeBuildInputs = [ makeWrapper copyDesktopItems jdk binutils ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    find -name "*.java" > sources.txt
+    javac @sources.txt
+    jar cvf dirbuster.jar .
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    export JAR=$out/share/java/${pname}/${pname}.jar
+    install -D ${pname}.jar $JAR
+    makeWrapper ${jre}/bin/java $out/bin/${pname} \
+      --add-flags "-classpath $JAR:''$CLASSPATH" \
+      --add-flags "-Xmx256M" \
+      --add-flags "com.sittinglittleduck.DirBuster.Start"
+
+    # Extract embedded desktop icon
+    mkdir -p $out/share/pixmaps
+    strings com/sittinglittleduck/DirBuster/ImageCreator.class | grep iVBORw0KG | base64 -d > $out/share/pixmaps/${pname}.png
+
+    # Symlink wordlists
+    mkdir -p $out/share/${pname}
+    for f in "${binaryDist}/share/${pname}/*"; do
+      ln -s $f $out/share/${pname}/
+    done
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Brute force directories and files names on web/application servers";
+    homepage = "http://www.owasp.org/index.php/Category:OWASP_DirBuster_Project";
```

Comment:
The url doesn't work for me, maybe https://wiki.owasp.org/index.php/Category:OWASP_DirBuster_Project is the right one?

Reply:
Cheers!